### PR TITLE
Add SAM 2 code snippet

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR adds a code snippet for SAM 2.

cc @haithamkhedr

### CODE REVIEW

**Review by julien-c — COMMENTED** [review]
there's already an open PR for this no?

**Comment by julien-c** [comment]
#831 from @merveenoyan

**Comment by merveenoyan** [comment]
@NielsRogge I have merged mine just now

**Comment by NielsRogge** [comment]
Thanks, sorry missed that. Renamed my PR to add a code snippet for SAM 2.

File: packages/tasks/src/model-libraries-snippets.ts [hunk 1]
```diff
@@ -507,6 +507,32 @@ export const fastai = (model: ModelData): string[] => [
...
+from sam2.sam2_video_predictor import SAM2VideoPredictor
+
+# for image prediction
+predictor = SAM2ImagePredictor.from_pretrained(${model.id})
+
+with torch.inference_mode(), torch.autocast("cuda", dtype=torch.bfloat16):
+    predictor.set_image(<your_image>)
+    masks, _, _ = predictor.predict(<input_prompts>)
+	
+# for video prediction
```
Comment (pcuenca) [inline packages/tasks/src/model-libraries-snippets.ts:522]: Maybe return an array so we have individual snippets for image and video that can be copied by the user?

**Review by pcuenca — APPROVED** [review]
Looks good to me, cc @merveenoyan in case she has additional feedback

File: packages/tasks/src/model-libraries-snippets.ts (hunk 1 above)
Comment (merveenoyan) [reply packages/tasks/src/model-libraries-snippets.ts:522]: we often put snippets to load them and not use them, no? this seems a bit overwhelming, the model cards have full examples already

**Review by Vaibhavs10 — COMMENTED** [review]
+1 to @pcuenca's comments about return Array's instead code ref: https://github.com/huggingface/huggingface.js/blob/473420604fa63ad0793b2da30b7a5ecb6bd6debf/packages/tasks/src/model-libraries-snippets.ts#L49

This is how it'd look: Audioseal ref: https://huggingface.co/facebook/audioseal?library=audioseal

**Review by pcuenca — APPROVED** [review]

**Review by Vaibhavs10 — APPROVED** [review]

File: packages/tasks/src/model-libraries-snippets.ts [hunk 2]
```diff
@@ -525,6 +525,33 @@ export const fastai = (model: ModelData): string[] => [
 learn = from_pretrained_fastai("${model.id}")`,
 ];
 
+export const sam2 = (model: ModelData): string[] => {
+	const image_predictor = `import torch
```
Comment (osanseviero) [inline packages/tasks/src/model-libraries-snippets.ts:529]: Maybe add a comment (just as in transformers) at the top of each snippet. E.g.

```suggestion
	const image_predictor = `# Use SAM2 with images
import torch
```

**Review by osanseviero — APPROVED** [review]
Very nice :fire: I left a minor comment but otherwise it looks good to me!

**Pull request merged**